Sources/SmithyTestUtil/ResponseTestUtil/FloatingPointEquivalenceChecker.swift

```diff
@@ -0,0 +1,24 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public func floatingPointValuesMatch(
+    lhs: (any FloatingPoint)?,
+    rhs: (any FloatingPoint)?
+) -> Bool {
```

Comment:
Rather than using `any`, I would make the signature generic.
```
public func floatingPointValuesMatch<FP: FloatingPoint>(lhs: FP?, rhs: FP?) -> Bool
```
This does two things:
1) It allows the compiler to resolve the types at compile time rather than run time.  This allows for better performance through generic specialization.
2) It requires that `lhs` and `rhs` are of the same type, which simplifies your logic within.  (Since you're using this to compare the same member in the same struct, that is always a valid condition.

Reply:
I almost forgot the most important part: with `lhs` and `rhs` constrained to be the same type, you can directly use Equatable to compare the two values rather than casting as you did below.

This enables the very simplified logic I posted below.

Reply:
(Equatable contains `Self` constraints, so you can only equate two values of the same type, even if they are both `FloatingPoint`s.)

Reply:
Changed as suggested 👍